### ISSUE TITLE
Consider dropping Python 4 upper bound

### ISSUE THREAD

**Issue by hauntsaninja** [PyCQA/isort#1986]
Any eventual Python 4 is unlikely to be extremely disruptive; isort will likely continue to work, whereas version capping makes it the case that isort definitely won't work with Python 4

My interest here is that any poetry projects that use isort (even just as a dev dependency) also end up needing to cap Python versions, causing these pessimistic assumptions to proliferate.

This section of https://iscinumpy.dev/post/bound-version-constraints/#pinning-the-python-version-is-special contains more details.

Would you be willing to entertain a PR that drops the upper bound here? https://github.com/PyCQA/isort/blob/12cc5fbd67eebf92eb2213b03c07b138ae1fb448/pyproject.toml#L43

**Comment by juhannc** [comment]
Python 4 is more than unlikely to happen [according to Guido van Rossum himself](https://youtu.be/aYbNh3NS7jA?t=2871) so I don't think it will be a problem.
It's probably better to watch out for new releases and see if and how they break `isort`.

**Comment by hauntsaninja** [comment]
Yes, dropping the upper bound is very much in line with the philosophy "watch out for new releases and see if and how they break isort", treating 4 like any other release instead of pessimistically assuming that it will break isort :-)

**Comment by staticdev** [comment]
@timothycrosley could you please create a new release with the changes we merged here https://github.com/PyCQA/isort/pull/1878? This will enable me to create a new PR to solve this.
